test/test_quantized_tensor.py

```diff
@@ -153,28 +155,58 @@ def quantize_c(data, scales, zero_points):
         self.assertTrue(np.allclose(r.numpy(), rqr.numpy(), atol=2 / np.min(scales.numpy())))
 
     def test_qtensor_permute(self):
-        r = torch.rand(100, 30, dtype=torch.float) * 2 - 4
-        scale = 2
-        zero_point = 2
+        r = torch.rand(10, 30, 2, 2, dtype=torch.float) * 4 - 2
+        scale = 0.02
+        zero_point = 1
         qr = torch.quantize_linear(r, scale, zero_point, torch.qint8)
         qr = qr.transpose(0, 1)
         rqr = qr.dequantize()
         # compare transpose + dequantized result with orignal transposed result
-        self.assertTrue(np.allclose(r.numpy().T, rqr.numpy(), atol=2 / scale))
+        self.assertTrue(np.allclose(r.numpy().transpose([1, 0, 2, 3]), rqr.numpy(), atol=2 / scale))
 
         qr = torch.quantize_linear(r, scale, zero_point, torch.qint8)
-        qr1 = qr.permute([1, 0])
+        qr1 = qr.permute([1, 0, 2, 3])
         qr2 = qr.transpose(0, 1)
         # compare int representation after transformations
-        self.assertTrue(torch.equal(qr1.int_repr(), qr2.int_repr()))
-        self.assertTrue(qr1.q_scale() == qr2.q_scale())
-        self.assertTrue(qr1.q_zero_point() == qr2.q_zero_point())
+        self.assertEqual(qr1.int_repr(), qr2.int_repr())
+        self.assertEqual(qr1.q_scale(), qr2.q_scale())
+        self.assertEqual(qr1.q_zero_point(), qr2.q_zero_point())
         # compare dequantized result
-        self.assertTrue(np.array_equal(qr1.dequantize().numpy(), qr2.dequantize().numpy()))
+        self.assertEqual(qr1.dequantize(), qr2.dequantize())
         # compare permuted + dequantized result with original transposed result
-        self.assertTrue(np.allclose(qr2.dequantize().numpy(), r.numpy().T, atol=2 / scale))
+        self.assertTrue(np.allclose(qr2.dequantize().numpy(), r.numpy().transpose([1, 0, 2, 3]), atol=2 / scale))
         # make permuted result contiguous
-        self.assertTrue(torch.equal(qr2.contiguous().int_repr(), qr2.int_repr()))
+        self.assertEqual(qr2.contiguous().int_repr(), qr2.int_repr())
+
+        # change memory format
+        qlast = qr.contiguous(memory_format=torch.channels_last)
+        self.assertEqual(qr.stride(), list(reversed(sorted(qr.stride()))))
+        self.assertNotEqual(qlast.stride(), list(reversed(sorted(qlast.stride()))))
+        self.assertEqual(qr.int_repr(), qlast.int_repr())
+        self.assertEqual(qr.q_scale(), qlast.q_scale())
+        self.assertEqual(qr.q_zero_point(), qlast.q_zero_point())
+        self.assertEqual(qlast.dequantize(), qr.dequantize())
+
+    def test_qtensor_per_channel_permute(self):
+        r = torch.rand(20, 10, 2, 2, dtype=torch.float) * 4 - 2
```

Comment:
Can we allow for permutes that do not change the axis that is quantized?
Would this work?
```python
# x is in nchw
i.e x_q = torch.quantize_linear_per_channel(... axis =[0])
# x_q is also in nchw
x_nhwc =  x_q.permute([0, 2, 3,1])
```

Reply:
as I mentioned in another place, we could, but I think we can add it later in a separate diff. And memory format works today already so it should be sufficient